docs/source/common/weights_loading.rst

```diff
@@ -4,32 +4,34 @@
     from pytorch_lightning.trainer.trainer import Trainer
     from pytorch_lightning.core.lightning import LightningModule
 
-.. _weights_loading:
+.. _checkpointing:
```

Comment:
just fyi: I'll be renaming this file to `checkpointing.rst` once I get all the required approvals. Not doing it now for easy review for reviewers. Doc tests might fail because references are already updated at other places.

Open to better filename suggestions though :)